findbugs/src/test/java/edu/umd/cs/findbugs/DetectorsTest.java

```diff
@@ -223,24 +228,16 @@ private void setUpEngine(String... analyzeMe) throws IOException {
             }
         }
 
-        // FIXME : the tests use our annotations, so we need them in the classpath...
-        String classpath = System.getProperty("java.class.path");
-        if (classpath == null) {
-            String rootDirectory;
-            try {
-                rootDirectory = new File(getClass().getResource("/").toURI()).getCanonicalPath();
-            } catch (final URISyntaxException e) {
-                throw new RuntimeException(e);
-            }
-            project.addAuxClasspathEntry(rootDirectory);
-        } else {
-            String[] cpParts = classpath.split(File.pathSeparator);
-            for (String cpStr : cpParts) {
-                File file = new File(cpStr);
-                if (file.isDirectory()) {
-                    project.addAuxClasspathEntry(file.getCanonicalPath());
+        File annotationsLib = new File("../spotbugs-annotations/build/libs");
```

Comment:
I'm not comfortable hardwiring this into the code. Specially when it's a build dir, and therefore a run from IDE may fail if the annotations project was cleaned from CLI.

What was wrong with the old code?